### PR TITLE
Add aws-infra Network implementation in Python

### PR DESCRIPTION
**Pull request by swgillespie**
This code came out of last Thursday's hackathon and serves as an example and test that we can write non-trivial components in Python.

This code is translated pretty much verbatim from the TypeScript implementation in this repo. All of this code is known to work with tip-of-master builds of `pulumi` and `pulumi-aws`, but we'll need to do another pass once the `v0.14` SDK drops.

Follow-up PRs will do the plumbing required to publish this package to PyPI.

### CODE REVIEW

**Comment by swgillespie** [comment]
I'm reviving this today. I'll push a new iteration in a little bit.

**Comment by swgillespie** [comment]
I think I've addressed all of the feedback from last week. I've tried to rework all of the code to adhere to `pylint` and the Google style guide for python. The code is all formatted using https://github.com/google/yapf.

File: python/pulumi_aws_infra/network.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def _create_from_arguments(self, name, number_of_availability_zones):
         :return: None
         """
         number_of_availability_zones = number_of_availability_zones or 2
-        if number_of_availability_zones < 1 or number_of_availability_zones > 4:
+        if number_of_availability_zones < 1 or number_of_availability_zones >= 4:
```
Comment (joeduffy) [inline python/pulumi_aws_infra/network.py]: This is driven by the AZs available in a region, right?  Or is there some other limitation here?

Comment (swgillespie) [reply python/pulumi_aws_infra/network.py]: Yes, that is true.

Comment (lukehoban) [reply python/pulumi_aws_infra/network.py]: Opened https://github.com/pulumi/pulumi-aws-infra/issues/21 to track potentially changing this to compute it from the target region.  It's not entirely trivial though, as we would then need to make an async call (which will likely in the future be a call that returns an `Output<T>`) before running the loop that creates each resource.  Interestingly, in Python that may not be as big a deal as it is in JavaScript - but I suspect we'll want to be consistent here across the two.

File: python/setup.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+setup(
+    name='pulumi_aws_infra',
+    version='${VERSION}',
+    description='Pulumi Amazon Web Services (AWS) infrastructure components.',
+    keywords='pulumi aws aws-infra',
+    url='https://github.com/pulumi/pulumi-aws-infra',
+    project_urls={'Repository': 'https://github.com/pulumi/pulumi-aws-infra'},
+    packages=find_packages(),
+    install_requires=[
+        'pulumi>=0.14.0,<0.15.0', 'pulumi_aws>=0.14'
```
[the remark this answers is not in the document]
Comment (swgillespie) [reply python/setup.py]: The `<0.15.0` is closest to the npm-style carat dependency. If we stick to our current versioning scheme, 0.15.0 won't be compatible with 0.14.0 and we'd be wrong to accept 0.15.0 as a dependency for this library.

I think we could relax this if we don't make any breaking changes to this library in 0.15, but I'm not sure we should relax this constraint until we are sure that will be the case.

**Comment by swgillespie** [comment]
I'm going to go ahead and merge this so we can start trying to hook up the publishing pipeline for this package - please let me know if there are any other changes anyone would like me to make.

**Comment by swgillespie** [comment]
thanks everyone for the reviews!

**Pull request merged**